db/data_migrations/20230606073111_migrate_profile_picture.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require "logger"
+require "redis"
+
+class MigrateProfilePicture < ActiveRecord::DataMigration
+  def up
+    log_file = Rails.root.join("log/profile_picture_migration.log")
+    custom_logger = Logger.new(log_file)
+    Rails.logger = custom_logger
+    redis = Redis.new
+    if redis.exists("last_migrated_user_id") == 1
+      last_migrated_user_id = redis.get("last_migrated_user_id").to_i
+      Rails.logger.info "Continuing migrating from user_id #{last_migrated_user_id + 1}"
+      migrate_paperclip_assets(last_migrated_user_id)
+    else
+      puts "Migrating User pfp to ActiveStorage & logging output to - #{log_file}"
+      Rails.logger.info "Total User profile_pictures to be migrated- #{User.where.not(profile_picture_file_name: nil).count}"
+      migrate_paperclip_assets
+    end
+  end
+
+  def migrate_paperclip_assets(last_migrated_user_id = 0)
+    redis = Redis.new
+    interrupted = false
+    trap("INT") do
+      interrupted = true
+      puts "Keyboard interrupt received. Saving progress and exiting gracefully..."
+    end
```

Comment:
Remove interrupt handling, it simply won't work when migration crashes due to OOM to any other errors.

Reply:
I had tested this part a lot, without this if we stop the migration then the objects won’t get uploaded to S3 nor the records are created as the migration raises ActiveRecord::Irreversible migration error so no records are saved

Reply:
Heres what happens when we stop the migration

![inter](https://github.com/CircuitVerse/CircuitVerse/assets/85568177/fb11769f-ecd3-4552-8ceb-4c3ebd2552a5)

No attachments present

![no att](https://github.com/CircuitVerse/CircuitVerse/assets/85568177/8f4ecf47-c8f6-41ec-bc9a-e3eb962d52aa)

We can remove those statements and do something like this, then it will work..

![int](https://github.com/CircuitVerse/CircuitVerse/assets/85568177/7721bcd8-5c00-496f-9be8-5a5e994cefa6)